src/efftester.ml

```diff
@@ -1175,13 +1316,27 @@ module Shrinker = struct
     | Float -> to_term (LitFloat (Gen.generate1 float.gen))
     | Bool -> to_term (LitBool (Gen.generate1 bool.gen))
     | String -> to_term (LitStr (Gen.generate1 StaticGenerators.string_gen))
-    | List _ | Fun _ | Typevar _ -> None
+    | Option _ | List _ | Fun _ | Typevar _ -> None
   ;;
 
   (* determines whether x occurs free (outside a binding) in the arg. exp *)
   let rec fv x = function
     | Lit _ -> false
     | Variable (_, y) -> x = y
+    | Constructor (_typ, _name, payload_lst) ->
+      let rec has_fv lst =
+        match lst with
+        | [] -> false
+        | trm :: rest -> if fv x trm then true else has_fv rest
+      in
+      has_fv payload_lst
```

Comment:
This is `List.exists (fv x) payload_list`; I would rename `payload_list` into `args` or just `ms` (the code uses `m, n` as variables for expressions/terms, so using `ms, ns` for lists of expressions makes sense).